src/Native/Runtime/amd64/InteropThunksHelpers.S

```diff
@@ -4,19 +4,27 @@
 
 .intel_syntax noprefix
 #include <unixasmmacros.inc>
-
-//TODO: Implement these stub method for non-windows
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+#define POINTER_SIZE 8
 
 LEAF_ENTRY RhpCommonStub, _TEXT
-    int 3
+    push_nonvol_reg rdi
+    mov    r11, [r10]
+    mov    rdi, r11
+    call   C_FUNC(SetThunkData)
```

Comment:
Since this is pretty performance sensitive path, we should do it without unnecessary overhead - do the TLS access directly like we do it in allocation helpers: https://github.com/dotnet/corert/blob/26413145abcbdd49468db230a6f29784494f6f46/src/Native/Runtime/unix/unixasmmacrosamd64.inc#L252